src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -3445,6 +3409,20 @@ private static void AddProviderEnumKind(ManifestBuilder manifest, FieldInfo stat
             eventData[eventAttribute.EventId].ActivityOptions = eventAttribute.ActivityOptions;
             eventData[eventAttribute.EventId].HasRelatedActivityID = hasRelatedActivityID;
             eventData[eventAttribute.EventId].EventHandle = IntPtr.Zero;
+
+            // We represent a byte[] as a integer denoting the length  and then a blob of bytes in the data pointer. This causes a spurious
+            // warning because eventDataCount is off by one for the byte[] case since a byte[] has 2 items associated it. So we want to check
+            // that the number of parameters is correct against the byte[] case, but also we the args array would be one too long if
+            // we just used the modifiedParamCount here -- so we need both.
```

Comment:
Nit: can you fix up the comment? It refers to the old variable name modifiedParamCount and the context of args array is no longer clear now that it isn't in context of the WriteToAllListeners() method

Reply:
Updated